crates/rattler_package_streaming/src/tokio/async_read.rs

```diff
@@ -30,17 +31,49 @@ pub async fn extract_tar_bz2(
 }
 
 /// Extracts the contents of a `.conda` package archive.
+/// This will perform on-the-fly decompression by streamign the reader.
 pub async fn extract_conda(
     reader: impl AsyncRead + Send + 'static,
     destination: &Path,
+) -> Result<ExtractResult, ExtractError> {
+    extract_conda_internal(
+        reader,
+        destination,
+        crate::read::extract_conda_via_streaming,
+    )
+    .await
+}
+
+/// Extracts the contents of a `.conda` package archive by first fully reading the streaming, instead
```

Comment:
maybe something like this:
`/// Extracts the contents of a `.conda` package archive by fully reading the streaming and then decompressing`

so basically we do not mention `on the fly-decompression`